src/app/sto/actions.js

```diff
@@ -91,10 +92,31 @@ export const configure = () => async (dispatch: Function, getState: GetState) =>
       contract.account,
     )
     dispatch(fetch())
+
+    const emailResult = await fetchAPI({
+      query: `
+        mutation ($accountData: AccountData!, $txHash: String!, $ticker: String!) {
+          withAccount(accountData: $accountData, txHash: $txHash) {
+            sendEmailSTOLaunched(ticker: $ticker)
```

Comment:
Why did you choose GraphQL and so complicated calls for this issue?

Reply:
And could you please describe and draw (if necessary) some diagrams on how your final solution works on the heroku side (with some references to polymath-offchain code)? What pros and cons it currently has?

Reply:
Added some information to the polymath-offchain readme: https://github.com/PolymathNetwork/polymath-offchain/blob/master/README.md

Reply:
Why do you send data like ticker or symbolDetails there? Why don't just grab it from the blockchain events on the API side? What prevents user to use our API to receive emails on transactions which are not related to the Polymath network?

Reply:
On retrieving the ticker and name from the blockchain events: That's interesting. I hadn't really thought about it. How would we do it for the STO email (an `addModule` transaction)? We can get the SecurityToken's address from the `to` field of the transaction but how do we get the ticker?

For validating the transactions, the server only checks that they're recent (last 2 minutes), sent by the the user and not used for a previous email. That makes sending an email cost at least as much as an Ethereum transaction so that there are much better ways to spam if someone wants to.

Thomas thinks a much bigger threat than spam is potential phishing attacks taking advantage of the fact that these emails come from Polymath and therefore look legitimate. He's made an Asana task to discuss those types of attacks but I can't seem to find it right now.

Thomas also informed me today that we _will_ be storing the user's email addresses (and verifying them) on the API so I think a lot of this will change. Verifying their email addresses helps prevent a lot of these attacks.